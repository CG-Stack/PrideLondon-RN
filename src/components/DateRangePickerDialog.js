// @flow
import React from "react";
import DateRangePicker from "./DateRangePicker";
import Dialog from "./Dialog";
import Text from "./Text";
import Touchable from "./Touchable";
import text from "../constants/text";
import type { DateRange } from "../data/date-time";
import { formatDateRange } from "../data/formatters";

type Props = {
  applyButtonText: string,
  dateRange: ?DateRange,
  onApply: () => void,
  onCancel: () => void,
  onChange: (?DateRange) => void,
  visible: boolean,
  forceNewRange: boolean
};

class DateRangePickerDialog extends React.PureComponent<Props> {
  clear = () => {
    this.props.onChange(null);
  };

  render() {
    const { dateRange, forceNewRange } = this.props;
    const title = dateRange
      ? formatDateRange(dateRange)
      : text.filterDayPickerTitle;

    return (
      <Dialog
        applyButtonText={this.props.applyButtonText}
        headerRight={
          <Touchable onPress={this.clear}>
            <Text>Clear</Text>
          </Touchable>
        }
        onApply={this.props.onApply}
        onCancel={this.props.onCancel}
        title={title}
        visible={this.props.visible}
      >
        <DateRangePicker
          onChange={this.props.onChange}
          dateRange={dateRange}
          forceNewRange={forceNewRange}
        />
      </Dialog>
    );
  }
}

export default DateRangePickerDialog;
