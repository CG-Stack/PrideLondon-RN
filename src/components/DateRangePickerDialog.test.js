// @flow
import React from "react";
import { shallow } from "enzyme";
import DateRangePickerDialog from "./DateRangePickerDialog";

it("renders correctly", () => {
  const output = shallow(
    <DateRangePickerDialog
      applyButtonText="Apply"
      dateRange={null}
      onApply={() => {}}
      onCancel={() => {}}
      onChange={() => {}}
      visible
    />
  );
  expect(output).toMatchSnapshot();
});
