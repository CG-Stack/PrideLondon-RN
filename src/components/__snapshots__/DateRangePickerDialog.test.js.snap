// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with date selection 1`] = `
<Dialog
  applyButtonText="Apply"
  headerRight={
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        markdown={false}
        style={Object {}}
        type="text"
      >
        Clear
      </Text>
    </TouchableOpacity>
  }
  onApply={[Function]}
  onCancel={[Function]}
  title="2 Feb -"
  visible={true}
>
  <DateRangePicker
    dateRange="2018-02-02"
    onChange={[Function]}
  />
</Dialog>
`;

exports[`renders correctly without date selection 1`] = `
<Dialog
  applyButtonText="Apply"
  headerRight={
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        markdown={false}
        style={Object {}}
        type="text"
      >
        Clear
      </Text>
    </TouchableOpacity>
  }
  onApply={[Function]}
  onCancel={[Function]}
  title="Select dates"
  visible={true}
>
  <DateRangePicker
    dateRange={null}
    onChange={[Function]}
  />
</Dialog>
`;
